fwb/app/(auth)/(routes)/fre2/fre2.tsx

```diff
@@ -0,0 +1,249 @@
+declare global {
+  interface Window {
+    Clerk: any;
+  }
+}
+
+export default function UserFlowPage2({ userData }: { userData: UserData }) {
+  const [company, setCompany] = useState("");
+  const [termsAndConditions, setTermsAndConditions] = useState("");
+  const [discountAmount, setDiscountAmount] = useState("");
+
+  //Set State of All Categories
+  const allCategories = [
+    "sports",
+    "fashion",
+    "electronic",
+    "health",
+    "books",
+    "hobbies",
+    "home & kitchen",
+    "computer & accessories",
+    "beauty & skincare",
+  ];
+  const [categories, setCategories] = useState<string[]>(allCategories);
+
+  //TODO: Handle User Routing Once Form is Submitted
+  const router = useRouter();
+
+  //Handle Discount Submission
+  const handleDiscountSubmit = async (e: any) => {
+    e.preventDefault();
+
+    try {
+      const bearerToken = await window.Clerk.session.getToken({
+        template: "testing_template",
+      });
+
+      const supabaseToken = await window.Clerk.session.getToken({
+        template: "supabase",
+      });
+
+      const formData = new FormData();
+      formData.append("company", company);
+      formData.append("terms_and_conditions", termsAndConditions);
+      formData.append("discount_amount", discountAmount);
+      formData.append("public", "true");
+      formData.append("categories", `${categories}`);
+      formData.append("company_url", `www.${company}.com`.toLowerCase());
+
+      // POST Fetch Request to Discounts API
+      const response = await fetch("/api/discounts", {
+        method: "POST",
+        headers: {
+          Authorization: `Bearer ${bearerToken}`,
+          supabase_jwt: supabaseToken,
+        },
+        body: formData,
+      });
+
+      // Log each entry in the FormData separately
+      formData.forEach((value, key) => {
+        console.log(`${key}: ${value}`);
+      });
+
+      if (response.ok) {
+        const data = await response.json();
+        console.log("Discount added successfully:", data);
+        router.push("/fre3");
+      } else {
+        const errorData = await response.json();
+        console.error("Error adding discount:", errorData);
+      }
+    } catch (error) {
+      console.error("Error using discount API:", error);
+    }
+  };
+
+  //Function to handle Array Property of Categories Column in Supabase
+  const handleCategoryChange = (selectedCategories: string[]) => {
+    // Set the state to the selected categories
+    if (selectedCategories.includes("All")) {
+      setCategories(allCategories);
+    } else {
+      setCategories(selectedCategories);
+    }
+  };
+
+  //Update user info based on provided company of Employment
+  const { isSignedIn, user, isLoaded } = useUser();
+
+  //Error handeling for if user tries to access page not signed in or Clerk isn't ready
+  useEffect(() => {
+    if (!isLoaded || !isSignedIn) {
+      return;
+    }
+
+    if (!userData || !userData.users[0].hasCompletedFRE) {
```

Comment:
I think this line of logic `!userData.users[0].hasCompletedFRE` is checking if `hasCompletedFRE` is false and if it is redirect them to /fre1. Currently, when I try to go through the fre process on a brand new email, when I hit fre2 it instantly redirects me back to fre1.  I believe that the comments you made about holding 3 variables for each page would be a possible solution.